src/ray/common/cgroup/README.md

```diff
@@ -0,0 +1,35 @@
+### Ray core cgroup documentation
+
+#### Physical execution mode
+
+Ray core supports a physical execution mode, which allows users to cap resource consumption for their applications.
+
+A few benefits:
+- It prevents application from eating up unlimited resource to starve other applications running on the same node;
+- It protects other processes on the node (including ray system components) to be killed due to insufficient resource (i.e. OOM);
+
+TODO(hjiang): reserve minimum resource will be supported in the future.
+
+#### Disclaimer and presumption
+
+- The feature is built upon cgroup, which only supports linux;
+- Only cgroup v2 is supported, meanwhile ray also requires application to have write permission and cgroup v2 be mounted in rw mode;
+- At the initial version, ray caps max resource usage via heuristric estimation (TODO: support user passed-in value).
+
+#### Implementation details
+
+cgroup v2 folders are created in tree structure as follows
+
+```
+        /sys/fs/cgroup/ray_<node_id>
+        /                          \
+.../system_cgroup           .../application_cgroup
+                            /                   \
+                .../default_cgroup/  .../<task_id>_<attempt_id>_cgroup (*N)
```

Comment:
[nit] task_default, and task_<task_id>_<attempt_id> ?

Reply:
In my latest commit I just call it `default`; do we need to differentiate task and actor here? Since from users' perspective they're different concepts, and cgroup is actually exposed to users.